packages/block-editor/src/components/block-preview/index.js

```diff
@@ -33,19 +37,57 @@ function BlockPreview( props ) {
 }
 
 export function BlockPreviewContent( { blocks, settings } ) {
+	const previewRef = useRef( null );
+
+	const [ previewScale, setPreviewScale ] = useState( 1 );
+	const [ visibility, setVisibility ] = useState( 'hidden' );
+
+	let timerId;
+
+	// Dynamically calculate the scale factor
+	useLayoutEffect( () => {
+		const { clientId } = blocks;
+		timerId = setTimeout( () => {
+			const refNode = previewRef.current;
+			const containerWidth = refNode.offsetWidth - ( 14 * 2 );
```

Comment:
What are these magic numbers `14` and `2`?

Reply:
Sorry, I removed the comments and forgot the clarify. We need to compute the offset of the container element. It's hardcoded for now.

Reply:
I've fixed this. Now dynamic.